src/simulators/density_matrix/densitymatrix_state.hpp

```diff
@@ -352,12 +352,8 @@ template <class densmat_t>
 size_t State<densmat_t>::required_memory_mb(uint_t num_qubits,
                                             const std::vector<Operations::Op> &ops)
                                             const {
-  // An n-qubit state vector as 2^n complex doubles
-  // where each complex double is 16 bytes
   (void)ops; // avoid unused variable compiler warning
```

Comment:
This is not part of your changes, but seeing this line, it makes clear that we need an override of this method.